pgjdbc/src/main/java/org/postgresql/jdbc/BatchResultHandler.java

```diff
@@ -221,7 +257,9 @@ public ResultSet getGeneratedKeys() {
     return newUpdateCounts;
   }
 
-  public int[] getUpdateCount() {
-    return uncompressUpdateCount();
+  public long[] getLargeUpdateCount() {
+    long[] copyCount = uncompressLongUpdateCount();
+    return Arrays.copyOf(copyCount, copyCount.length);
```

Comment:
Is copy required here?

Reply:
It's not really required, I guess there are no side effects of exposing the original array, so I will change that part.